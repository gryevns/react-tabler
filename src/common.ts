export type Color =
    | "blue"
    | "azure"
    | "indigo"
    | "purple"
    | "pink"
    | "red"
    | "orange"
    | "yellow"
    | "lime"
    | "green"
    | "teal"
    | "cyan"
    | "gray";

export const colors = [
    "blue" as const,
    "azure" as const,
    "indigo" as const,
    "purple" as const,
    "pink" as const,
    "red" as const,
    "orange" as const,
    "yellow" as const,
    "lime" as const,
    "green" as const,
    "teal" as const,
    "cyan" as const,
    "gray" as const,
];

export type Variant =
    | "primary"
    | "secondary"
    | "success"
    | "info"
    | "warning"
    | "danger";

export const variants = [
    "primary" as const,
    "secondary" as const,
    "success" as const,
    "info" as const,
    "warning" as const,
    "danger" as const,
];

export type Size = "sm" | "md" | "lg" | "xl";

export const sizes = ["sm" as const, "md" as const, "lg" as const, "xl" as const];
