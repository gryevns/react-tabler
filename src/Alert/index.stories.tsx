import React from "react";
import { Bell } from "react-feather";

import Alert from "./index";
import Buttons from "../Buttons";
import { variants } from "../common";

export default {
    title: "Alert",
};

export const Variants = () =>
    variants.map((variant) => (
        <Alert variant={variant}>This is a {variant} alert - check it out!</Alert>
    ));

export const Dismissible = () => (
    <>
        <Alert variant="warning" dismiss>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </Alert>
        <Alert variant="warning" dismiss={false}>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </Alert>
    </>
);

export const Icon = () => (
    <Alert variant="info">
        <Bell size={18} className="mr-1" /> You have done 5 actions.
    </Alert>
);

export const Children = () => (
    <Alert variant="success">
        <h4>Some Message</h4>
        <p>
            Lorem ipsum Minim ad pariatur eiusmod ea ut nulla aliqua est quis id
            dolore minim voluptate.
        </p>
        <Buttons.List>
            <Buttons.Button variant="success">Okay</Buttons.Button>
            <Buttons.Button variant="secondary">No, thanks</Buttons.Button>
        </Buttons.List>
    </Alert>
);
