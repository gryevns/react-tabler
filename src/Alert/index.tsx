import React from "react";
import cn from "classnames";
import { Variant } from "../common";

export interface AlertProps {
    children: React.ReactNode;
    variant: Variant;
    className?: string;
    dismiss?: boolean;
    style?: React.CSSProperties;
    visible?: boolean;
}

const Alert = ({
    children,
    className,
    dismiss = true,
    style,
    variant,
    visible = true,
}: AlertProps) => {
    const [isVisible, setVisible] = React.useState(visible);
    if (!isVisible) {
        return null;
    }
    const classes = cn(
        "alert",
        `alert-${variant}`,
        {
            "alert-dismissible": dismiss,
        },
        className,
    );
    const onClick = (e: React.MouseEvent) => {
        setVisible(false);
        e.preventDefault();
    };
    return (
        <div className={classes} style={style} role="alert">
            {dismiss && (
                <a
                    href="#"
                    className="close"
                    data-dismiss="alert"
                    aria-label="close"
                    onClick={onClick}
                >
                    &times;
                </a>
            )}
            {children}
        </div>
    );
};

export default Alert;
