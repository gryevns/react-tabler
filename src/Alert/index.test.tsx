import React from "react";
import { shallow } from "enzyme";

import { variants, Variant } from "../common";
import Alert from "./index";

describe("Alert", () => {
    it("should match variant snapshot", () => {
        variants.map((variant: Variant) => {
            const wrapper = shallow(<Alert variant={variant}>Content</Alert>);
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should render null if invisible", () => {
        const wrapper = shallow(
            <Alert variant="primary" visible={false}>
                Hidden
            </Alert>,
        );
        expect(wrapper.isEmptyRender()).toBe(true);
    });
    it("should hide on close", () => {
        const wrapper = shallow(<Alert variant="primary">Content</Alert>);
        expect(wrapper.isEmptyRender()).toBe(false);
        wrapper.find("a").simulate("click", { preventDefault: () => {} });
        expect(wrapper.isEmptyRender()).toBe(true);
    });
});
