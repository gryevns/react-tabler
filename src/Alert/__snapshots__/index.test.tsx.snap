// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match variant snapshot 1`] = `
<div
  className="alert alert-primary alert-dismissible"
  role="alert"
>
  <a
    aria-label="close"
    className="close"
    data-dismiss="alert"
    href="#"
    onClick={[Function]}
  >
    ×
  </a>
  Content
</div>
`;

exports[`Alert should match variant snapshot 2`] = `
<div
  className="alert alert-secondary alert-dismissible"
  role="alert"
>
  <a
    aria-label="close"
    className="close"
    data-dismiss="alert"
    href="#"
    onClick={[Function]}
  >
    ×
  </a>
  Content
</div>
`;

exports[`Alert should match variant snapshot 3`] = `
<div
  className="alert alert-success alert-dismissible"
  role="alert"
>
  <a
    aria-label="close"
    className="close"
    data-dismiss="alert"
    href="#"
    onClick={[Function]}
  >
    ×
  </a>
  Content
</div>
`;

exports[`Alert should match variant snapshot 4`] = `
<div
  className="alert alert-info alert-dismissible"
  role="alert"
>
  <a
    aria-label="close"
    className="close"
    data-dismiss="alert"
    href="#"
    onClick={[Function]}
  >
    ×
  </a>
  Content
</div>
`;

exports[`Alert should match variant snapshot 5`] = `
<div
  className="alert alert-warning alert-dismissible"
  role="alert"
>
  <a
    aria-label="close"
    className="close"
    data-dismiss="alert"
    href="#"
    onClick={[Function]}
  >
    ×
  </a>
  Content
</div>
`;

exports[`Alert should match variant snapshot 6`] = `
<div
  className="alert alert-danger alert-dismissible"
  role="alert"
>
  <a
    aria-label="close"
    className="close"
    data-dismiss="alert"
    href="#"
    onClick={[Function]}
  >
    ×
  </a>
  Content
</div>
`;
